mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/RetentionConfiguration.java

```diff
@@ -0,0 +1,68 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.vault;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.configuration.Configuration;
+import org.apache.james.util.TimeConverter;
+
+import com.google.common.base.Preconditions;
+
+public class RetentionConfiguration {
+    public static final RetentionConfiguration DEFAULT = new RetentionConfiguration(Duration.ofDays(365));
```

Comment:
Do you mean 1 year ? (if so, as you may know, a year is not always 365 days)
There's an awesome `Years` class in org.threeten.extra

Reply:
Well, DAY is a time unit that don't really depend on the period it applies. IMO 365 days is satisfying enough.

Reply:
Well, saying 365 is a year is an approximation. Why don't you want to use the method I suggested?

Reply:
I implemented this support as part of TimeConverter, relying on ChronoUnit